### ISSUE TITLE
Define photoscans

### ISSUE THREAD

**Issue by sadhana-r** [OpenwaterHealth/OpenLIFU-python#177]
Add to openlifu the concept of a photoscan and the ability to serialize them to and from json. Photoscans are added to a particular session, similar to how volumes are added to a subject. Similar to how volumes metadata include the filepath to the mri file, photoscan metadata will include the filepath to the model file, mtl file, and texture file. It will also include an approval flag with indicates whether the scan is good enough for use in transducer tracking. This issue will add a method to the Database class to write/load a photoscan for a particular session. The Database should also be able to retrieve photoscans by ID.

This assumes that the photoscan <> mri transform is saved to sessions file and that a single photoscan won't be used across different sessions.

**Comment by ebrahimebrahim** [comment]
The mtl file we probably wouldn't need, just obj and an exr/png would do. because the mtl is material properties

**Comment by ebrahimebrahim** [comment]
> Would there be multiple photoscans associated with a session?

Yes, there could be I think. If the photoscan isn't good enough then one may want to acquire a new one in the same session.

> Should only be one approved photoscan, but there may be multiple "unapproved" scans?

Right, pretty much. I guess there's no problem with there being multiple approved ones, so no need to enforce "only one approved"

> This assumes that the photoscan <> mri transform is saved to sessions file and that a single photoscan won't be used across different sessions.

This part is up in the air still ... let's talk more about this.

**Comment by ebrahimebrahim** [comment]
A good model to follow for photoscans+sessions is volumes+subjects, in that
- each subject has a subfolder of volumes
- each volume is a json file of metadata, part of which is a relative filepath pointing to the actual 3D volume data file in the same directory
- the database provides methods to load the volume metadata into a dictionary or openlifu python object of some kind, and when it's loaded the relative filepath becomes an absolute filepath pointing to the volume data file
- the database provides methods to write a volume, which writes the json metadata and which simply copies the volume data file into place.

So translating that to photoscans+sessions:
- each session has a subfolder of photoscans
- each photoscan is a json file of metadata, part of which is relative filepaths pointing to the obj and texture data files in the same directory
- the database provides methods to load the photoscan metadata into a dictionary or openlifu python object of some kind, and when it's loaded the relative filepaths become absolute filepaths pointing to the obj and texture files
- the database provides methods to write a photoscan, which writes the json metadata and which simply copies the obj and exr files into place.

Notably omitted here is any way of loading the photoscan data into memory as a mesh+texture.
We do want to add the ability to do that in openlifu python, but I propose doing that in a separate issue and as spearate functionality from Database. Maybe openlifu.io.photoscans or something like that. LMK if you agree with this design or what do you think
